prometheusmetrics.go

```diff
@@ -35,11 +41,15 @@ func NewPrometheusProvider(r metrics.Registry, namespace string, subsystem strin
 		Registry:         r,
 		promRegistry:     promRegistry,
 		FlushInterval:    FlushInterval,
-		gauges:           make(map[string]prometheus.Gauge),
 		customMetrics:    make(map[string]*CustomCollector),
 		histogramBuckets: []float64{0.05, 0.1, 0.25, 0.50, 0.75, 0.9, 0.95, 0.99},
 		timerBuckets:     []float64{0.50, 0.95, 0.99, 0.999},
 		mutex:            new(sync.Mutex),
+
+		// initialize config with gauges and labels
+		gauges:     make(map[string]*prometheus.GaugeVec),
+		labels:     make(prometheus.Labels),
```

Comment:
I think it is better to add a labels parameter so that users can pass some initial lables in. Think of this case, in a proccess, we new 2 sarama producers, one named "primary", another one named "standby" and we want to seperate the metrics of these 2 producers, it better that we have a "name" label, or may be we can't seperate them.